src/process/shell.cr

```diff
@@ -0,0 +1,105 @@
+class Process
+  # Converts a sequence of strings to one joined string with each argument shell-quoted.
+  #
+  # This is then safe to pass as part of the command when using `shell: true` or `system()`.
+  #
+  # NOTE: The actual return value is system-dependent, so it mustn't be relied on in other contexts.
+  # See also: `quote_posix`.
+  #
+  # ```
+  # files = ["my file.txt", "another.txt"]
+  # `grep -E 'fo+' -- #{Process.quote(files)}`
+  # ```
+  def self.quote(args : Enumerable(String)) : String
+    {% if flag?(:win32) %}
+      quote_windows(args)
+    {% else %}
+      quote_posix(args)
+    {% end %}
+  end
+
+  # Shell-quotes one item, same as `quote({arg})`.
+  def self.quote(arg : String) : String
+    quote({arg})
+  end
```

Comment:
```suggestion
  # Shell-quotes given items, same as `quote(*args)`.
  def self.quote(*args : String) : String
    quote(args)
  end
```

Reply:
You can be sure that I considered this option, it's not like I didn't know about it.

But I'm wary about the effect of proliferating these splats on the compilation times.

And I also want to stress the case of quoting *one* item as something that one may need specifically. If there are multiple items, it's nice to make the user think, is this a List situation or is it a Tuple situation.

I don't know. I don't feel that strongly about this though.